lib/xcodeproj/project/object/build_configuration.rb

```diff
@@ -102,15 +107,25 @@ def resolve_build_setting(key, root_target = nil)
 
         private
 
+        VARIABLE_NAME_PATTERN =
+          '( # capture block
+            [_a-zA-Z0-9]+? # non-greedy lookup for everything contained in this list
+          )'.freeze
+        private_constant :VARIABLE_NAME_PATTERN
+
         CAPTURE_VARIABLE_IN_BUILD_CONFIG = /
             \$ # matches dollar sign literally
-            [{(] # matches a single caracter on this list
-              ( # capture block
-              [^inherited] # ignore if match characters in this list
-              [$(){}_a-zA-Z0-9]*? # non-greedy lookup for everything that contains this list
-              )
-            [})] # matches a single caracter on this list
+            (?: # non-capturing group
+              [{] # matches a single caracter on this list
```

Comment:
nit: typo `character` here. also `.freeze` this?

Reply:
regexps are already immutable